app/user.go

```diff
@@ -2273,15 +2273,39 @@ func (a *App) PromoteGuestToUser(user *model.User, requestorId string) *model.Ap
 
 	for _, team := range userTeams.Data.([]*model.Team) {
 		// Soft error if there is an issue joining the default channels
-		if err := a.JoinDefaultChannels(team.Id, user, false, requestorId); err != nil {
+		if err = a.JoinDefaultChannels(team.Id, user, false, requestorId); err != nil {
 			mlog.Error(fmt.Sprintf("Encountered an issue joining default channels err=%v", err), mlog.String("user_id", user.Id), mlog.String("team_id", team.Id), mlog.String("requestor_id", requestorId))
 		}
 	}
+
+	promotedUser, err := a.GetUser(user.Id)
+	if err != nil {
+		return err
+	}
+
+	message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_USER_UPDATED, "", "", "", nil)
+	message.Add("user", promotedUser)
```

Comment:
Should probably make a const for this `user` key as it's being used everywhere and can lead to problems if there is an accidental typo. Also good for consistency so people don't have to hunt around to see what's the key used to notify update on a `user` model.

If it increases the scope of the change, I would recommend adding a `TODO` or perhaps a help-wanted ticket, so next time someone comes around using this key they can help do the refactor. 

Thoughts?

Reply:
Yes, I like the idea, but not only is out of the scope of the PR, this PR goes to a feature branch for guest accounts, so probably the worse place to add more changes.

I like it but I'm not 5/5 about it, because we are talking about the keys in the object sent through the websocket, for now that are 28 different possible keys that we can put in constants, anyway, we can screw the thing using the wrong constant, probably would make more sense to model the X kind of events that we want to return as independent structs.

Reply:
True, would make sense to model it out, I was thinking in the interim, since modelling it out would take some time.

Ah didn't see that it was going into a feature branch. 
Make sense in that case 👍